drivers/serial/uart_nrf5.c

```diff
@@ -196,7 +136,7 @@ static int baudrate_set(struct device *dev,
  */
 static int uart_nrf5_init(struct device *dev)
 {
-	volatile struct _uart *uart = UART_STRUCT(dev);
+	NRF_UART_Type *uart = UART_STRUCT(dev);
 	struct device *gpio_dev;
```

Comment:
Binding of this device might fail but this is not checked (and should be at least with an assertion).
I can't place a comment on line 143.